src/domain/eventSearch/searchPanel/SearchPanel.tsx

```diff
@@ -0,0 +1,171 @@
+import { css } from 'emotion';
+import { IconHeart, IconLocation, Koros } from 'hds-react';
+import React from 'react';
+import { useTranslation } from 'react-i18next';
+import { useHistory, useLocation } from 'react-router';
+
+import Button from '../../../common/components/button/Button';
+import MultiSelectDropdown from '../../../common/components/multiSelectDropdown/MultiSelectDropdown';
+import SearchInput from '../../../common/components/searchInput/SearchInput';
+import { ROUTES } from '../../../constants';
+import useLocale from '../../../hooks/useLocale';
+import { OptionType } from '../../../types';
+import Container from '../../app/layout/Container';
+import FormContainer from '../../app/layout/FormContainer';
+import { useTheme } from '../../app/theme/Theme';
+import useEventTypeOptions from '../../event/hooks/useEventTypeOptions';
+import {
+  getEventSearchInitialValues,
+  getEventSearchQuery,
+} from '../../eventSearch/utils';
+import FilterSummary from '../filterSummary/FilterSummary';
+import DateSelectorDropdown, {
+  DATE_FIELDS,
+} from './dateSelectorDropdown/DateSelectorDropdown';
+import PlaceSelector from './placeSelector/PlaceSelector';
+import styles from './searchPanel.module.scss';
+
+const SearchPanel: React.FC = () => {
+  const { theme } = useTheme();
+  const { t } = useTranslation();
+  const history = useHistory();
+  const location = useLocation();
+  const locale = useLocale();
+
+  const eventTypeOptions = useEventTypeOptions();
+
+  const [searchValue, setSearchValue] = React.useState('');
+  const [startDate, setStartDate] = React.useState<Date | null>(null);
+  const [endDate, setEndDate] = React.useState<Date | null>(null);
+  const [selectedPlaces, setSelectedPlaces] = React.useState<string[]>([]);
+  const [selectedEventTypes, setSelectedEventTypes] = React.useState<string[]>(
+    []
+  );
+
+  const handleSearch = (text: string) => {
+    history.push({
+      pathname: `/${locale}${ROUTES.SEARCH}`,
+      search: getEventSearchQuery({
+        end: endDate,
+        place: selectedPlaces,
+        start: startDate,
+        text,
+        type: selectedEventTypes,
+      }),
+    });
+  };
+
+  const handleChangePlaces = (newPlaces: OptionType[]) => {
+    setSelectedPlaces(newPlaces.map((item) => item.value));
+  };
+
+  const handleChangeEventTypes = (newTypes: OptionType[]) => {
+    setSelectedEventTypes(newTypes.map((item) => item.value));
+  };
+
+  const handleChangeDate = (field: DATE_FIELDS, value: Date | null) => {
+    switch (field) {
+      case DATE_FIELDS.END_DATE:
+        setEndDate(value);
+        break;
+      case DATE_FIELDS.START_DATE:
+        setStartDate(value);
+        break;
+    }
+  };
+
+  React.useEffect(() => {
+    const { end, places, start, text, types } = getEventSearchInitialValues(
+      location.search
+    );
+    setEndDate(end);
+    setStartDate(start);
+    setSelectedPlaces(places);
+    setSearchValue(text);
+    setSelectedEventTypes(types);
```

Comment:
This component could use useReducer to mimick state as one object. That might make life easier here.

`const [state, setState] = React.useReducer((s, a) => ({ ...s, ...a }), defaultState)`

```
setState({
  end,
  start,
  places,
  ....
})
```

Just an idea, not a must change of course

Reply:
And of course with some typings it is event better

Reply:
This is really nice suggestion. I didn't know that I can use useReducer that way. 

Changed